sources/org.osbuild.skopeo

```diff
@@ -103,17 +98,43 @@ class SkopeoSource(sources.SourceService):
                            check=True)
 
             # Verify that the digest supplied downloaded the correct container image id.
-            # The image id is the digest of the config, but skopeo can' currently
+            # The image id is the digest of the config, but skopeo can't currently
             # get the config id, only the full config, so we checksum it ourselves.
             res = subprocess.check_output(["skopeo", "inspect", "--raw", "--config", destination])
             downloaded_id = "sha256:" + hashlib.sha256(res).hexdigest()
             if downloaded_id != image_id:
                 raise RuntimeError(
                     f"Downloaded image {imagename}@{digest} has a id of {downloaded_id}, but expected {image_id}")
 
-            # Atomically move download dir into place on successful download
+            # fetch the manifest and merge it into the archive
+            self.merge_manifest(source, dest_path, extra_args)
```

Comment:
This function should only run when the digest points to a manifest list.  Otherwise, we're doing extra work for no reason.  Running `skopeo copy --multi-arch=index-only` with a single image digest will pull the image, so we end up copying the same archive twice and duplicating the manifest in the merged directory.  Is that right @mtrmac?

Reply:
Ugh, yes, `--multi-arch=index-only` with a non-multi-arch image currently copies everything. It’s clearly documented that way, but it’s also rather unhelpful for this use case. (OTOH the fix would be to _fail_ with non-multi-arch, which I suppose, doesn’t help :) )

Either way, that leads to a question of “how does a caller tell if an image is multi-arch”, and the _good_ answer to that is an unimplemented RFE https://github.com/containers/skopeo/issues/1283  . Without that, it’s possible, although not at all elegant, to decide that outside of Skopeo using `inspect --raw` and inspecting that JSON manually.

Reply:
> Without that, it’s possible, although not at all elegant, to decide that outside of Skopeo using inspect --raw and inspecting that JSON manually.

Yeah that was my plan now.  Inspect it and look at the type.